### PR TITLE
statd: add support for ietf-interfaces/interface/admin-status

### PR DESCRIPTION
**Pull request by troglobit**
Xlate IFF_UP flag to admin-status: up/down.  We do not support any 'testing' mode (yet) in Infix.

### CODE REVIEW

File: src/statd/statd.c [hunk 1]
```diff
@@ -391,6 +391,31 @@ static int ly_add_ip_link_br(const struct ly_ctx *ctx, struct lyd_node **parent,
 	return SR_ERR_OK;
 }
 
+static int iface_has_flag(json_t *j_iface, const char *ifflag)
+{
+	json_t *flags, *flag;
+	int found = 0;
```
Comment (rical) [inline src/statd/statd.c:397]: I don't see the need for this flag? Why not just return 1 when you find the flag?
Something like:
```
if (strcmp(json_string_value(j_flag), flag) == 0)
    return 1
```

File: src/statd/statd.c [hunk 2]
```diff
@@ -391,6 +391,31 @@ static int ly_add_ip_link_br(const struct ly_ctx *ctx, struct lyd_node **parent,
...
+
+	flags = json_object_get(j_iface, "flags");
+	if (!flags)
+		return 0;
+
+	if (!json_is_array(flags))
+		return 0;
+
+	json_array_foreach(flags, i, flag) {
+		ERROR("Comparing flag %s vs ifflag %s", json_string_value(flag), ifflag);
```
Comment (rical) [inline src/statd/statd.c:408]: Accidental debug left in the code?

File: src/statd/statd.c [hunk 3]
```diff
@@ -391,6 +391,31 @@ static int ly_add_ip_link_br(const struct ly_ctx *ctx, struct lyd_node **parent,
 	return SR_ERR_OK;
 }
 
+static int iface_has_flag(json_t *j_iface, const char *ifflag)
```
Comment (rical) [inline src/statd/statd.c:394]: If you want to align the variable naming to the rest of the statd code, this should be `const char *flag` and the json flags below should be `j_flags`and `j_flag`. No big deal.

**Review by rical — COMMENTED** [review]
Minor style comments

File: src/statd/statd.c (hunk 2 above)
Comment (troglobit) [reply src/statd/statd.c:408]: Ah, yes. Will fix.

File: src/statd/statd.c (hunk 3 above)
Comment (troglobit) [reply src/statd/statd.c:394]: OK, will fix!

**Review by rical — APPROVED** [review]

**Pull request merged**